### DISCUSSION TITLE
How to get stdpath("data") in coc-settings.json?

### DISCUSSION THREAD

**Discussion by hh02** [neoclide/coc.nvim#4645] [Q&A] [answered]
How to get stdpath("data") in coc-settings.json?

**Comment by hh02** [comment]
Does coc-settings.json has any predefined variables？
Just like vscode: [vscode variable reference](https://code.visualstudio.com/docs/editor/variables-reference#_predefined-variables)

**Reply by fannheyward** [reply]
Yes, this variables are supported in coc.nvim.

**Comment by fannheyward** [comment, marked as answer]
coc.nvim didn't support to expand `stdpath` for now.

You can use `env`:

1. define something like `VIM_STDPATH_DATA`
2. use `env:VIM_STDPATH_DATA` in your coc-settings.json, coc.nvim will expand this to the real value.

**Comment by chemzqm** [comment]
`:h coc-configuration-expand`
